src/ParallelAlgorithms/Interpolation/Actions/InterpolationTargetReceiveVars.hpp

```diff
@@ -124,11 +129,24 @@ struct InterpolationTargetReceiveVars {
       return;
     }
 
-    InterpolationTarget_detail::add_received_variables<InterpolationTargetTag>(
-        make_not_null(&box), vars_src, global_offsets, temporal_id);
+    if (not vars_have_already_been_received) {
+      InterpolationTarget_detail::add_received_variables<
+          InterpolationTargetTag>(make_not_null(&box), vars_src, global_offsets,
+                                  temporal_id);
+    }
     if (InterpolationTarget_detail::have_data_at_all_points<
             InterpolationTargetTag>(box, temporal_id)) {
       // All the valid points have been interpolated.
+
+      // Check if functions of time are ready on this component
+      if (not InterpolationTarget_detail::functions_of_time_are_ready<
+              InterpolationTargetReceiveVars>(
+              cache, array_index,
+              std::add_pointer_t<ParallelComponent>{nullptr}, temporal_id,
+              vars_src, global_offsets, temporal_id, true)) {
```

Comment:
You never use `vars_src` in the action this calls, and I'm guessing it could be a large object that is getting copied.  Perhaps send an empty vector or something?